tarxx.h

```diff
@@ -101,20 +104,102 @@ namespace tarxx {
 
         void add_file(const std::string& filename)
         {
+            if (stream_file_header_pos_ >= 0) throw std::logic_error("Can't add new file before add_file_complete has been called");
```

Comment:
I guess the the exception text should read `stream_file_complete` (instead of `add_file_complete`). I would suggest rephrasing to have it a little less technical.
```suggestion
            if (stream_file_header_pos_ >= 0) throw std::logic_error("Can't add new file while adding streaming data isn't completed");
```

Reply:
done